lib/addressable/uri.rb

```diff
@@ -768,6 +776,55 @@ def self.form_unencode(encoded_value)
       end
     end
 
+    ##
+    # Resolves paths to their simplest form.
+    #
+    # @param [String] path The path to normalize.
+    #
+    # @return [String] The normalized path.
+    def self.normalize_path(path)
+      # Section 5.2.4 of RFC 3986
+
+      return nil if path.nil?
+      normalized_path = path.dup
+      begin
+        mod = nil
+        mod ||= normalized_path.gsub!(RULE_2A, SLASH)
```

Comment:
you can make this a regular assignment and delete the line above

Reply:
@luke-hill actually - no. We can't change it coz of not obvious `end until mod.nil?` here: https://github.com/sporkmonger/addressable/pull/319#diff-33998b1f6fff21cf34d63816274234b3R809

I checked it locally, unfortunately, we get an infinite loop without this line.

However, this is not my code, actually, It became from the previous authors (igrigorik), while looks short becomes unreadable. In this PR I just moved it.
Anyway, thank you for your time.

Reply:
sorry, silly me. you are right. fixed it. Thank you